velox/type/tests/TypeTest.cpp

```diff
@@ -22,10 +22,7 @@ using namespace facebook::velox;
 
 namespace {
 void testTypeSerde(const TypePtr& type) {
-  velox::DeserializationRegistryForSharedPtr().Register(
-      Type::getClassName(),
-      static_cast<std::shared_ptr<const Type> (*)(const folly::dynamic&)>(
-          Type::create));
+  Type::registerSerDe();
```

Comment:
I was expecting here a test that actually passes some context and verifies that you can use while deserializing stuff. Is this coming in the next PR?

Reply:
This is a "stacked PR" and this change comes from https://github.com/facebookincubator/velox/pull/4306.

I added a unit test for 'context' in deserialize.